feat!: Better platform independent sizes

Feichtmeier opened the pull request:
Since the VisualDensity API is kinda incomplete, I made a custom solution with `bool get isMobile =>
    !kIsWeb && (Platform.isAndroid || Platform.isIOS || Platform.isFuchsia);` 

Web is excluded because when compared to for example github, the compact widgets fit much better.
While on it I discovered a lot if themes we did not include yet :laughing: 
I also discovered :monocle_face: IconButton.outlined, .tonal and .filled
Which have a flipped color but this is a different topic for a different PR and switching the green to filled will prbly break a lot of apps, so let's not do this now.

However the compact sizes should fit us very well but I guess we need to test this a bit.

Some screenshots...

Since the colors are untouched I only show the dark theme:

| Comfortable | Compact |
| -| -|
|![grafik](https://github.com/ubuntu/yaru.dart/assets/15329494/a33d09c8-bbc4-45d8-9f4c-325f4ceb6f4d)|![grafik](https://github.com/ubuntu/yaru.dart/assets/15329494/a7434d14-921f-433d-81dd-097ac092fc8a)|
|![grafik](https://github.com/ubuntu/yaru.dart/assets/15329494/ef679701-4534-4c48-ae45-0322b66336f8)|![grafik](https://github.com/ubuntu/yaru.dart/assets/15329494/17f62a12-6236-40d8-b22b-a1406fe8a527)|



Fixes #369

Jupi007 reviewed (commented):
At a first look, it seems good.
But I'm curious to see how it will look with Yaru icons now 🤔

Jupi007 commented:
Oh, it looks like our icons are pixel perfect in desktop mode, and blurry in mobile mode:

![Capture d’écran du 2023-08-10 00-19-22](https://github.com/ubuntu/yaru.dart/assets/36476595/ea9b9597-0f00-4cfc-a573-dde8ada475f3)
![image](https://github.com/ubuntu/yaru.dart/assets/36476595/c1fc16e5-4068-46a2-bf39-3416d3758714)


Which is fine!

Feichtmeier commented:
That's strange because I think 22 is the default size xD

Need to test this PT myself in real apps but honestly I really don't use buttons so much anymore xD

I think the app that will be affected most are @jpnurmi 's quiver of installer apps

Feichtmeier commented:
Okay, instead of making compact icons super tiny I changed the content padding for iconbuttons if it is not mobile.
I think this is less dangerous and looks better by default. updated the screenshots

Jupi007 commented on lib/src/themes/constants.dart:
```diff
@@ -1,8 +1,13 @@
 const kCheckRadius = 3.0;
 const kWindowRadius = 8.0;
 const kAppBarElevation = 0.0;
+const kCompactAppBarHeight = 47.0;
```
Same comment as the button size, I know we use this value since a while, but it would be better to use an even value here (46.0 or 48.0).

Jupi007 commented on lib/src/themes/constants.dart:
```diff
@@ -1,8 +1,13 @@
+const kComfortableAppBarHeight = 56.0;
 
-const kDisabledGreyDark = Color(0xFF535353);
+const kCompactButtonHeight = 35.0;
```
We shouldn't use odd size values, because it results in sub-pixel alignment (7.5 here):

![Capture d’écran du 2023-08-15 10-59-52](https://github.com/ubuntu/yaru.dart/assets/36476595/57a2e441-355a-412f-86d2-9c1a2975d9bf)

It would be better to use 34.0 or 36.0:

![Capture d’écran du 2023-08-15 10-59-13](https://github.com/ubuntu/yaru.dart/assets/36476595/75e1a3aa-8e5d-43cb-b736-f8255f80e3b1)

Feichtmeier replied:
interesting, well, I guess the 1px should not be too annoying for the appbar I guess

on the iconsize, before you start... what is the actual icon size of yaru/adwaita in gtk? I tried to look it up in the gtk4 theme but 36 can really not be

Jupi007 replied:
> interesting, well, I guess the 1px should not be too annoying for the appbar I guess

IMO we can use +1px without problem on buttons and appbar.

> on the iconsize, before you start... what is the actual icon size of yaru/adwaita in gtk? I tried to look it up in the gtk4 theme but 36 can really not be

Gtk is using 16x16 by default. That's the size of the canvas of every yaru symbolic icons.

Feichtmeier replied:
yes +1px to buttons and titlebars sounds like a plan :+1: 
eventually I can do some border trick with boxshadow lets see
the only mode where you will notice the 1px is in a split screen anyways

on the icon size.... 16px  is something different in flutter then
![grafik](https://github.com/ubuntu/yaru.dart/assets/15329494/07782765-a42a-4f76-a591-61d524029c90)
if I want to reach size size inside an icon button with 0 padding I need something around 20 in flutter

Jupi007 replied:
> eventually I can do some border trick with boxshadow lets see

Can't we do 46px + 1px border, so it looks similar to Gtk, but with an even inner size? 🤔

> on the icon size.... 16px is something different in flutter then

Well... gtk icons use the full canvas, while Flutter ones are 20x20 on a 24x24 canvas 😅
So when you resize a Flutter icon to 20x20, the icon draw itself becomes something like 16x16.

Feichtmeier replied:
aha, okay let's try 20 then. Also Maybe we should just use yaru icons in the example :shrug: 
do not really care so much for chunky material icons
ohhhhh I am an idiot
it is actually possible to set the strokalign to -1
result:
![grafik](https://github.com/ubuntu/yaru.dart/assets/15329494/c11b9c0e-a453-45ca-a3bd-81993c4ae5ce)
with height 46
should the buttons then be 34 or 36?

Jupi007 replied:
> aha, okay let's try 20 then. Also Maybe we should just use yaru icons in the example shrug
> do not really care so much for chunky material icons

+1 :+1: 

> with height 46
> should the buttons then be 34 or 36?

Well, 34 imo. We need to reduce both sizes to make them fit.

Jupi007 reviewed (changes requested):
I have a few comments about odd sizes.

I'm also really annoyed by the icon rendering at 20px...

![image](https://github.com/ubuntu/yaru.dart/assets/36476595/de89b507-4855-420c-8a66-902ca2f75d5c)

I did some tests, and it is possible to create an icon library optimized for 20x20 icons:

![image](https://github.com/ubuntu/yaru.dart/assets/36476595/b4da00cd-b187-4ec3-9a65-6baf21e0f7df)

Feichtmeier commented:
nice idea, pushed

Feichtmeier commented:
also I think we should use jet for foreground colors in the light theme

Jupi007 reviewed (approved):
I didn't noticed anything else.
LGTM :+1:
